packages/accordion/src/useAccordion.ts

```diff
@@ -5,185 +5,150 @@
 import {
-  composeEventHandlers,
   KEY_CODES,
-  getControlledValue
+  composeEventHandlers,
+  getControlledValue,
+  useId
 } from '@zendeskgarden/container-utilities';
 
-export interface IUseAccordionProps {
-  /** Prefixes IDs for the accordion trigger and panels  */
-  idPrefix?: string;
-  /** Sets the expanded sections in a controlled accordion */
-  expandedSections?: number[];
-  /** Sets the default expanded sections in a uncontrolled accordion */
-  defaultExpandedSections?: number[];
-  /** Handles accordion expansion changes */
-  onChange?: (expanded: number) => any;
-  /** Determines if multiple panels can be expanded at the same time in an uncontrolled accordion */
-  expandable?: boolean;
-  /** Determines if panels can be collapsed in an uncontrolled accordion */
-  collapsible?: boolean;
-}
-
-interface IHeaderProps extends HTMLProps<any> {
-  ariaLevel?: number | null;
-  role?: any;
-}
-
-interface ITriggerProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-  tabIndex?: any;
-}
-
-interface IPanelProps extends HTMLProps<any> {
-  index?: number;
-  role?: any;
-}
-
-export interface IUseAccordionPropGetters {
-  getHeaderProps: <T>(options?: T & IHeaderProps) => any;
-  getTriggerProps: <T>(options?: T & ITriggerProps) => any;
-  getPanelProps: <T>(options?: T & IPanelProps) => any;
-}
-
-export interface IUseAccordionReturnValue extends IUseAccordionPropGetters {
-  expandedSections: number[];
-  disabledSections: number[];
-}
+import {
+  IUseAccordionProps,
+  IUseAccordionReturnValue,
+  IHeaderProps,
+  ITriggerProps,
+  IPanelProps
+} from './types';
 
 export function useAccordion({
   idPrefix,
+  sections,
   expandedSections,
+  defaultExpandedSections,
   onChange,
   expandable = true,
-  collapsible = true,
-  defaultExpandedSections
+  collapsible = true
 }: IUseAccordionProps = {}): IUseAccordionReturnValue {
-  const isControlled = expandedSections !== null && expandedSections !== undefined;
-  const seed = useUIDSeed();
-  const prefix = useMemo<string>(
-    () => idPrefix || seed(`accordion_${PACKAGE_VERSION}`),
-    [idPrefix, seed]
-  );
+  const prefix = useId(idPrefix);
   const TRIGGER_ID = `${prefix}--trigger`;
   const PANEL_ID = `${prefix}--panel`;
+
+  const isControlled = expandedSections !== null && expandedSections !== undefined;
   const [expandedState, setExpandedState] = useState<number[]>(defaultExpandedSections || [0]);
+  const [disabledState, setDisabledState] = useState(collapsible ? [] : expandedState);
+  const internalExpandedState = getControlledValue(expandedSections, expandedState)!;
 
-  const controlledExpandedState = getControlledValue(expandedSections, expandedState)!;
+  const toggle = useCallback(
+    (index: number) => {
+      const expanded: number[] = [];
+      const disabled: number[] = [];
 
-  const [disabledState, setDisabledState] = useState(collapsible ? [] : expandedState);
+      sections!.forEach(sectionIndex => {
```

Comment:
`sections` doesn't appear to be a required prop; this feels like more reason to enforce it as required, to help match other containers using this pattern. :)

Reply:
Updated to make this required!